compiler/noirc_evaluator/src/ssa/opt/assert_constant.rs

```diff
@@ -169,3 +211,112 @@ fn append_foreign_call_param(
         panic!("ICE: expected constant value");
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::{assert_ssa_snapshot, errors::RuntimeError, ssa::ssa_gen::Ssa};
+
+    #[test]
+    fn do_not_fail_on_assert_constant_in_empty_loop() {
+        let src = r"
+        acir(inline) fn main f0 {
+          b0():
+            v2 = call f1() -> [Field; 0]
+            v4, v5 = call as_slice(v2) -> (u32, [Field])
+            v6 = make_array [] : [Field]
+            v7 = allocate -> &mut u32
+            store u32 0 at v7
+            v9 = allocate -> &mut [Field]
+            store v6 at v9
+            jmp b1(u32 0)
+          b1(v0: u32):
+            v10 = lt v0, u32 0
+            jmpif v10 then: b2, else: b3
+          b2():
+            v13 = lt v0, u32 0
+            constrain v13 == u1 1
+            v15 = load v7 -> u32
+            v16 = load v9 -> [Field]
+            call assert_constant(v2)
+            v19, v20 = call slice_push_back(v15, v16) -> (u32, [Field])
+            store v19 at v7
+            store v20 at v9
+            v22 = unchecked_add v0, u32 1
+            jmp b1(v22)
+          b3():
+            v11 = load v7 -> u32
+            v12 = load v9 -> [Field]
+            return
+        }
+        brillig(inline) fn foo f1 {
+          b0():
+            v0 = make_array [] : [Field; 0]
+            return v0
+        }
```

Comment:
Seems like we can minimise this more. A lot of the code is unrelated to what we're testing.

Reply:
I will remove the unneeded code.